### PR TITLE
Handle parameter names corresponding to vectors

### PR DESCRIPTION
**Pull request by costachris**
In preparation for function learning we'll need to seamlessly deal with parameter names that correspond to vectors. This is handled here by treating each vector component as a unique parameter in EKP.

### CODE REVIEW

**Comment by costachris** [comment]
Still double-checking this for  `julia_parallel`...

File: src/DistributionUtils.jl [hunk 1]
```diff
@@ -39,16 +39,50 @@ function construct_priors(
...
     # All vars are approximately uniform in unconstrained space
     distributions = repeat([Parameterized(Normal(0.0, unconstrained_σ))], n_param)
     to_file ? jldsave(joinpath(outdir_path, "prior.jld2"); distributions, constraints, u_names) : nothing
     return ParameterDistribution(distributions, constraints, u_names)
 end
 
+"""
+    flatten_param_dict(param_dict::Dict{String, Vector{Constraint}})
+
+For parameter names that correspond to vectors, assign unqiue name to each vector component.
```
Comment (ilopezgp) [inline src/DistributionUtils.jl:61]:
```suggestion
For parameter names that correspond to vectors, assign unique name to each vector component.
```

File: src/DistributionUtils.jl [hunk 2]
```diff
@@ -39,16 +39,50 @@ function construct_priors(
...
+    constraints :: vector of constraints
+"""
+function flatten_param_dict(param_dict::Dict{String, Vector{Constraint}})
+
+    u_names = Vector{String}()
+    constraints = Vector{Vector{Constraint}}()
+    for (param, value) in param_dict
+        if length(value) > 1
+            for j in 1:length(value)
+                append!(u_names, [string(param, "_{", string(j), "}")])
```
Comment (ilopezgp) [inline src/DistributionUtils.jl:75]:
```suggestion
                push!(u_names, "$(param)_{$j}")
```

File: src/DistributionUtils.jl [hunk 3]
```diff
@@ -39,16 +39,50 @@ function construct_priors(
...
+                append!(constraints, [[value[j]]])
```
Comment (ilopezgp) [inline src/DistributionUtils.jl:76]: Same here, if you are appending elements, you can use `push!`,

```suggestion
                push!(constraints, [value[j]])
```

**Review by ilopezgp — COMMENTED** [review]
nice solution!

**Review by ilopezgp — APPROVED** [review]

**Comment by ilopezgp** [comment]
Closes #100

**Pull request merged**